### ISSUE TITLE
Parse C Source does not recognize function pointer array in struct definition

### ISSUE THREAD

**Issue by madebr** [NationalSecurityAgency/ghidra#3908]
**Describe the bug**
The C Parser ignores a function pointer array used in a struct.

**To Reproduce**
1. Create the following dummy header named e.g. **dummy.h**:
    ```c
    typedef struct SomeStruct {
        int first_member;
        int (*procArray1[2])(int*, int*);
        int anotherMember;
        int (*procArray2[2])(int*, int*);
        int (*LoneProc1)(char, int);
        int (*LoneProc2)(char, int);
        int finalMember;
    } SomeStruct;
    ```
2. Start Ghidra -> CodeBrowser -> FIle -> Parse C Source
3. Clear all source files to parse + clear all parse options
4. Add the **dummy.h** to the Source files to parse list
5. Click on "Parse to File..."
6. Choose to store the data type archive (=gdt) somewhere + click ok.
7. In the Data Type Manager, open the just-created gdt (=click on down arrow in right-upper corner -> Open File Archive... -> choose the gdt that you created in step 6)
8. Open the SomeStruct datatype in the Structure Editor (you need to open the archive in edit mode for this).

`procArray1` and `procArray2` are not present in the struct.procArray1.
The parser has correctly parsed the function definitions (_func_1 and _func_2),
but does not use them in the SomeStruct.

**Expected behavior**
`_func_1*[2] ProcArray1` and `_func_2*[2] ProcArray2` are present in the struct.

**Screenshots**
Selected options in the Parse C Source dialog
![Screenshot from 2022-01-24 03-30-32](https://user-images.githubusercontent.com/4138939/150712509-0be758ee-a5ea-4ba7-bf23-2096c847bbf5.png)

The Structure layout (missing procArray1 and procArray2):
![Screenshot from 2022-01-24 05-48-15](https://user-images.githubusercontent.com/4138939/150723809-2cd9e2ab-d170-4aa6-9e5a-8959c0601686.png)

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

**Environment (please complete the following information):**
 - OS: Fedora Linux 34
 - Java Version: 11.0.13
 - Ghidra Version: 10.1.1
 - Ghidra Origin: official

**Comment by emteere** [comment]
Will be fixed in an upcoming set of changes.

**Comment by ryanmkurtz** [comment]
Fixed by e42c1f1f3263e192d5d9042c14f1f25413931aa0
